File: src/IsbInterop/Data/Wrappers/PickRequisite.vb
Namespace Data.Wrappers

    ''' <summary>
    ''' Обертка над IRequisite.
    ''' </summary>
    Friend Class PickRequisite
        Inherits Requisite
        Implements IPickRequisite

        ''' <summary>
        ''' Список описаний допустимых значений реквизита.
        ''' </summary>
        Public ReadOnly Property Items As IPickRequisiteItems Implements IPickRequisite.Items
            Get
                Dim rcwItems = Me.GetRcwProperty("Items")
                Return New PickRequisiteItems(rcwItems, Scope)
            End Get
        End Property

        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="rcwRequisite">COM-объект реквизит.</param>
        ''' <param name="scope">Область видимости.</param>
        Friend Sub New(rcwRequisite As Object, scope As IScope)
            MyBase.New(rcwRequisite, scope)
        End Sub

    End Class
End Namespace

File: src/IsbInterop.CSharp/Data/Wrappers/PickRequisite.cs
namespace IsbInterop.Data.Wrappers
{
  /// <summary>
  /// Обертка над IRequisite.
  /// </summary>
  internal class PickRequisite : Requisite, IPickRequisite
  {
    /// <summary>
    /// Список описаний допустимых значений реквизита.
    /// </summary>
    public IPickRequisiteItems Items
    {
      get
      {
        var rcwItems = this.GetRcwProperty("Items");
        return new PickRequisiteItems(rcwItems, this.Scope);
      }
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwRequisite">COM-объект реквизит.</param>
    /// <param name="scope">Область видимости.</param>
    internal PickRequisite(object rcwRequisite, IScope scope) : base(rcwRequisite, scope) { }
  }
}
